### PR TITLE
Handle system UUID retrieval for Power(ppc64) Systems

### PR DESCRIPTION
**Pull request by bpradipt**
This patch adds requisite support to retrieve system uuid details for Power
systems. Power systems do not have DMI data. However most of the relevant details
are either in /proc or /sys. For baremetal servers, the UID is available in
/proc/device-tree/system-id. For guests the UUID is available in
/proc/device-tree/vm,uuid inside the guest. Guest's /proc filesystem do not have
/proc/device-tree/system-id

Example
On baremetal system
$cat /proc/device-tree/system-id
2122AAA

On a guest VM
$cat /proc/device-tree/vm,uuid
4b1a1a7e-079e-479c-8072-d8108f31050c

Signed-off-by: Pradipta Kr. Banerjee <bpradip@in.ibm.com>

### CODE REVIEW

**Comment by vmarmol** [comment]
ok to test

**Comment by bpradipt** [comment]
Hello I'm working in IBM and covered as part of Google Corporate CLA. Is there any additional info required?

**Comment by vmarmol** [comment]
I see the corporate CLA, thanks! The bot doesn't know how to check corporate CLA (unless you setup a group for it).

**Comment by vmarmol** [comment]
btw:

The following files are not properly formatted:
manager/machine.go

File: utils/sysfs/sysfs.go [hunk 1]
```diff
@@ -233,7 +235,20 @@ func (self *realSysFs) GetCacheInfo(id int, name string) (CacheInfo, error) {
 }
 
 func (self *realSysFs) GetSystemUUID() (string, error) {
-	id, err := ioutil.ReadFile(path.Join(dmiDir, "id", "product_uuid"))
+	var uuidPath string
+
+	if strings.Contains(runtime.GOARCH, "ppc64") {
```
Comment (vmarmol) [inline utils/sysfs/sysfs.go]: I'd say the same thing here that @rjnagal suggested. Let's just check all of these and take the first one that provides a UUID

Comment (rjnagal) [inline utils/sysfs/sysfs.go]: +1

Comment (bpradipt) [inline utils/sysfs/sysfs.go]: Thanks @vmarmol. I'll resend incorporating the suggestion.

**Comment by bpradipt** [comment]
@vmarmol, I have uploaded new version based on your suggestions. Can you please review.
Thanks

**Comment by rjnagal** [comment]
LGTM

**Pull request merged**